### PR TITLE
planner: add a switch control if prepare stmt with param limit can be cached 

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #40219 

Problem Summary:

### What is changed and how it works?
Add a switch `tidb_enable_plan_cache_for_param_limit` control plan with param limit cacheable, the switch is turned on by default, if some one encountered problems about this, this can be a workaround.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cacheable_checker.go [hunk 1]
```diff
@@ -135,22 +135,24 @@ func (checker *cacheableChecker) Enter(in ast.Node) (out ast.Node, skipChildren
...
+				return in, true
+			}
+		}
+		if node.Offset != nil {
+			if _, isParamMarker := node.Offset.(*driver.ParamMarkerExpr); isParamMarker {
+				checker.cacheable = false
+				checker.reason = "query has 'limit ?, 10' is un-cacheable"
+				return in, true
+			}
+		}
```
Comment (qw4990) [inline planner/core/plan_cacheable_checker.go:155]: How about:
```
		if node.Count != nil {
			if _, isParamMarker := node.Count.(*driver.ParamMarkerExpr); isParamMarker && !checker.sctx.GetSessionVars().EnablePlanCacheForParamLimit {
				checker.cacheable = false
				checker.reason = "query has 'limit ?' is un-cacheable"
				return in, true
			}
		}
		if node.Offset != nil {
			if _, isParamMarker := node.Offset.(*driver.ParamMarkerExpr); isParamMarker && !checker.sctx.GetSessionVars().EnablePlanCacheForParamLimit {
				checker.cacheable = false
				checker.reason = "query has 'limit ?, 10' is un-cacheable"
				return in, true
			}
		}
```

**Review by qw4990 — APPROVED** [review]
LGTM

**Review by Reminiscent — APPROVED** [review]

**Pull request merged**